src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -278,19 +278,20 @@ def get_vm_cloud_type(self):
         for i in range(0, Constants.MAX_IMDS_CONNECTION_RETRY_COUNT):
             try:
                 res = urlreq.urlopen(request, timeout=2)
-                print(res.get_code())
+                print("Return code from IMDS connection http request: {0}.".format(str(res.getcode())))
                 if(res.getcode() == 200):
                     print("Connection to IMDS end point successfully established. VMCloudType is Azure\n")
                     return Constants.VMCloudType.AZURE
                 else:
                     raise
-            except:
+            except Exception as error:
                 """ Failed to connect to Azure IMDS endpoint. This is expected on Arc machine - but not expected on Azure machine."""
+                print('Exception from IMDS connection http request: ' + repr(error))
+                print("Failed to connect to IMDS end point. [Retry Count={0}].".format(str(i)))
```

Comment:
Retry Count=0 looks may raise some queries like it did for me. Change this to 
print("Failed to connect to IMDS end point. [Trial ={0}].".format(str(i+1)))

So that should show up in the first attempt as
"Failed to connect to IMDS end point. [Trial =1] this will go all the way upto 4

Log for the last attempt, under else, is good

Some existing logs for reference: 
![image](https://user-images.githubusercontent.com/56841542/130690375-de7490f4-4a82-4ea7-a15f-719b7a8e7449.png)